client/jetpack-cloud/sections/partner-portal/issue-multiple-licenses-form/index.tsx

```diff
@@ -1,21 +1,29 @@
 import { Button } from '@automattic/components';
 import { getQueryArg } from '@wordpress/url';
 import { useTranslate } from 'i18n-calypso';
-import { useCallback, useState, useEffect } from 'react';
+import { useCallback, useEffect } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import { useIssueMultipleLicenses } from 'calypso/jetpack-cloud/sections/partner-portal/hooks';
 import LicenseBundleCard from 'calypso/jetpack-cloud/sections/partner-portal/license-bundle-card';
 import LicenseProductCard from 'calypso/jetpack-cloud/sections/partner-portal/license-product-card';
-import {
-	isJetpackBundle,
-	selectAlphaticallySortedProductOptions,
-} from 'calypso/jetpack-cloud/sections/partner-portal/utils';
+import TotalCost from 'calypso/jetpack-cloud/sections/partner-portal/primary/total-cost';
+import { isJetpackBundle } from 'calypso/jetpack-cloud/sections/partner-portal/utils';
 import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import useProductsQuery from 'calypso/state/partner-portal/licenses/hooks/use-products-query';
 import {
 	getAssignedPlanAndProductIDsForSite,
 	hasPurchasedProductsOnly,
 } from 'calypso/state/partner-portal/licenses/selectors';
+import {
+	addSelectedProductSlugs,
+	clearSelectedProductSlugs,
+	removeSelectedProductSlugs,
+} from 'calypso/state/partner-portal/products/actions';
```

Comment:
The selected products are used to calculate the total cost which is displayed in the `TotalCost` component so they have to be available outside of this component. I have never seen state extracted to a hook which is used in multiple places. Wouldn't that just create separate instances of the state, equivalent to calling `useState` in different components with the name being the same?

Reply:
Oops, I actually meant using `useContext()` and the context API if we meant to share it between components.

Reply:
But don't feel obligated to change it - I was just offering alternatives you could experiment with.

Reply:
Oh I understand now. I'm inclined to go with `redux` over `useContext()` since we already have `redux` managing other state in Calypso and I like consistency. Usually I would go for `useContext()` in order to avoid importing `redux` to an app if it's not fully needed, but if it's already there I figure it's nice to have all the state managed in as few places as possible.

Reply:
Sounds good to me 👍 As long as we know a few options and we evaluated what works best in our context (no pun intended), we should be good.